aopy/analysis/base.py

```diff
@@ -867,3 +867,72 @@ def interp_multichannel(x):
+    edge conditions and replaces input NaN values with 0 to calculate the correlation map. After the correlation map is calculated,
+    all values were NaN in the input data are again set to NaN.
+
+    Args:
+        data1 (nrow, ncol): First input data array. Used as baseline if map alignment is required.
+        data2 (nrow, ncol): Second input data array. Shifted to match the baseline if map alignment is required
+        knlsz (int): Length of the kernel window in units of data points. The kernel is a square so each side will have the lenght specified here. This value should always be odd.
+        align_maps (bool): Whether or not to align maps.
+
+    Returns:
+        tuple: Tuple containing:
+            | **NCC (nrow, ncol):** Spatial correlation map
+            | **shifts (tuple):** Contains (row_shifts, col_shifts)
+    '''
+    
+    NCC = np.zeros((data1.shape))
+    data_sz = data1.shape[0]
+    
+    # Get nan value locations 
+    nan_idx1 = np.isnan(data1)
+    nan_idx2 = np.isnan(data2)
+    
+    # Replace NaNs with 0s so correlation doesn't output NaN
+    data1[nan_idx1] = 0
+    data2[nan_idx2] = 0
+    
+    # Get maxidx of 2D spatial correlation matrix to ensure data maps are aligned.
+    if align_maps:
+        corr = scipy.signal.correlate2d(data1/np.linalg.norm(data1), data2/np.linalg.norm(data2), 
+                                        boundary='fill', mode='same')
+        irow, icol = np.unravel_index(np.argmax(corr), corr.shape)  # find the match
+        row_shift = int(irow - (data1.shape[0]-1)/2)
+        col_shift = int(icol - (data1.shape[1]-1)/2)
+        data2_align = np.roll(data2, row_shift, axis=0)
+        data2_align = np.roll(data2_align, col_shift, axis=1)
+        shifts = (row_shift, col_shift)
+    else:
+        data2_align = data2
+        shifts = None
+    
+    # Pad data
+    data1_pad = np.pad(data1, int((knlsz-1)/2), mode='constant')
+    data2_pad = np.pad(data2_align, int((knlsz-1)/2), mode='constant')
+    
+    start_idx = int((knlsz-1)/2)
+    end_idx = int(data_sz + (knlsz-1)/2)
+    for xx in range(start_idx, end_idx):
+        for yy in range(start_idx,end_idx):
+            # Normalize input arrays based on the norm
+            data_subset1 = data1_pad[(xx-start_idx):(xx+start_idx+1),(yy-start_idx):(yy+start_idx+1)]
+            data_subset2 = data2_pad[(xx-start_idx):(xx+start_idx+1),(yy-start_idx):(yy+start_idx+1)]
+
+            # If either data subset is all 0's set the NCC to 0
+            if np.linalg.norm(data_subset1)==0 or np.linalg.norm(data_subset2)==0:
+                NCC[xx - start_idx, yy - start_idx]
+            else:
+                data_subset1 /= np.linalg.norm(data_subset1)
+                data_subset2 /= np.linalg.norm(data_subset2)
+                NCC[xx - start_idx, yy - start_idx] = np.max(scipy.signal.correlate2d(data_subset1,data_subset2))
```

Comment:
not sure i understand the np.max here, could you explain in comments?

Reply:
scipy.signal.correlate2d computes the cross correlation between the two arrays input. I couldn't find a python function to just compute the 2d correlation value. We should probably be taking correlation at 0 lag though.